App crashing while uploading certain pictures.

**Summary:** 
I tried to upload one picture the app crashed. But it's not happening with every picture that I try to upload. Happens with picture located in content://com.android.providers.downloads.documents/document/

**Steps to reproduce:** 
Open app and click on upload picture and try to upload picture(Does not happen with every picture).

**Add System logs:**
```
03-17 03:06:15.693 748-748/fr.free.nrw.commons.debug E/AndroidRuntime: FATAL EXCEPTION: main
Process: fr.free.nrw.commons.debug, PID: 748
java.lang.RuntimeException: Unable to start activity ComponentInfo{fr.free.nrw.commons.debug/fr.free.nrw.commons.upload.ShareActivity}: java.lang.NullPointerException: Attempt to get length of null array
at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2659)
at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2724)
at android.app.ActivityThread.-wrap12(ActivityThread.java)
 at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1473)
at android.os.Handler.dispatchMessage(Handler.java:102)
 at android.os.Looper.loop(Looper.java:154)
 at android.app.ActivityThread.main(ActivityThread.java:6123)
at java.lang.reflect.Method.invoke(Native Method)
at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:867)
at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:757)
Caused by: java.lang.NullPointerException: Attempt to get length of null array
at fr.free.nrw.commons.upload.ShareActivity.findOtherImages(ShareActivity.java:537)
at fr.free.nrw.commons.upload.ShareActivity.getFileMetadata(ShareActivity.java:519)
at fr.free.nrw.commons.upload.ShareActivity.performPreUploadProcessingOfFile(ShareActivity.java:398)
at fr.free.nrw.commons.upload.ShareActivity.onCreate(ShareActivity.java:298)
at android.app.Activity.performCreate(Activity.java:6672)
at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1140)
at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2612)
at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2724) 
at android.app.ActivityThread.-wrap12(ActivityThread.java) 
at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1473) 
at android.os.Handler.dispatchMessage(Handler.java:102) 
at android.os.Looper.loop(Looper.java:154) 
 at android.app.ActivityThread.main(ActivityThread.java:6123) 
 at java.lang.reflect.Method.invoke(Native Method) 
 at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:867) 
 at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:757) 
03-17 03:06:16.375 748-1001/fr.free.nrw.commons.debug D/ExistingFileAsync: File already exists in Commons: false


```

**Expected behavior:** 
It should not crash.

**Observed behavior:** 
It crashed.
![device-2018-03-16-194758](https://user-images.githubusercontent.com/17095817/37526049-739ef586-2954-11e8-91e2-837aeec0e209.gif)


**Device and Android version:** 
Moto G4 plus 
Android 7.0.0
Xiaomi
 8.1.0

 **Commons app version:** 
2.6.7


**Would you like to work on the issue?**
Yes i would like to work on this issue.

--- a/app/src/main/java/fr/free/nrw/commons/upload/FileUtils.java
+++ b/app/src/main/java/fr/free/nrw/commons/upload/FileUtils.java
@@ -59,7 +59,7 @@ public class FileUtils {
 
                 final String id = DocumentsContract.getDocumentId(uri);
                 final Uri contentUri = ContentUris.withAppendedId(
-                        Uri.parse("content://downloads/public_downloads"), Long.valueOf(id));
+                        Uri.parse("content://downloads/document"), Long.valueOf(id));
 
                 returnPath =  getDataColumn(context, contentUri, null, null);
             } else if (isMediaDocument(uri)) { // MediaProvider

